llama_index/embeddings/dashscope.py

```diff
@@ -0,0 +1,331 @@
+    async def _aget_text_embedding(self, query: str) -> List[float]:
+        """Get text embedding."""
+        return self._get_text_embedding(query)
+
+    # TODO: user proper async methods
+    async def _aget_query_embedding(self, query: str) -> List[float]:
+        """Get query embedding."""
+        return self._get_query_embedding(query)
+
+
+class DashScopeBatchTextEmbedding:
+    """DashScope class for batch text embedding.You can input a large amount of text to be
+       embedding into a file line by line, upload it to the network (it needs to be
+       publicly accessible), and call the embedding service through the file URL.
+       After the task is completed, the embedding result is saved on the network
+       and the result file URL is returned to you.
+
+    Args:
+        model (str): Model for embedding.
+            Defaults to DashScopeBatchTextEmbeddingModels.TEXT_EMBEDDING_ASYNC_V2.
+                Options are:
+
+                - DashScopeBatchTextEmbeddingModels.TEXT_EMBEDDING_ASYNC_V1
+                - DashScopeBatchTextEmbeddingModels.TEXT_EMBEDDING_ASYNC_V2
+        api_key (str): The DashScope api key.
+    """
+
+    _api_key: Optional[str] = PrivateAttr()
+    _model: str = PrivateAttr()
+
+    def __init__(
+        self,
+        model: str = DashScopeBatchTextEmbeddingModels.TEXT_EMBEDDING_ASYNC_V2,
+        api_key: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        self._api_key = api_key
+        self._model = model
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "DashScopeBatchTextEmbedding"
+
+    def get_batch_query_embedding(self, embedding_file_url: str) -> Optional[str]:
+        """Get batch query embeddings.
+
+        Args:
+            embedding_file_url (str): The url of the file to embedding which with lines of text to embedding.
+
+        Returns:
+            str: The url of the embedding result, format ref:
+                 https://help.aliyun.com/zh/dashscope/developer-reference/text-embedding-async-api-details.
+        """
+        return get_batch_text_embedding(
+            self._model,
+            embedding_file_url,
+            api_key=self._api_key,
+            text_type=DashScopeTextEmbeddingType.TEXT_TYPE_QUERY,
+        )
+
+    def get_batch_text_embedding(self, embedding_file_url: str) -> Optional[str]:
+        """Get batch text embeddings.
+
+        Args:
+            embedding_file_url (str): The url of the file to embedding which with lines of text to embedding.
+
+        Returns:
+            str: The url of the embedding result, format ref:
+                 https://help.aliyun.com/zh/dashscope/developer-reference/text-embedding-async-api-details.
+        """
+        return get_batch_text_embedding(
+            self._model,
+            embedding_file_url,
+            api_key=self._api_key,
+            text_type=DashScopeTextEmbeddingType.TEXT_TYPE_DOCUMENT,
+        )
+
+
+class DashScopeMultiModalEmbedding:
```

Comment:
Multimodal embeddings should probably go in their own `dashscope_multimodal.py` file, and they should extend the `MultiModalEmbedding` class

Reply:
updated.

Reply:
I noticed this class is still here -- did you maybe forget to add the new file when committing?